momepy/preprocessing.py

```diff
@@ -1101,3 +1105,453 @@ def roundabout_simplification(
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "euclidean":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _euclidean_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    elif method == "spider":
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+    else:
+        print("Simplification method not recognized. Using spider-web simplification.")
+        edges_simplified_geometries = edges_tosimplify_gdf.apply(
+            lambda edge: _spider_simplification(
+                edge.geometry, edge.new_origin_pt, edge.new_destination_pt, buffer
+            ),
+            axis=1,
+        )
+        edges_simplified_gdf = edges_tosimplify_gdf.assign(
+            new_geometry=edges_simplified_geometries
+        )
+
+    # Rename and update the columns:
+    cols_rename = {
+        edge_from_col: "original_from",
+        edge_to_col: "original_to",
+        "origin_cluster": edge_from_col,
+        "destination_cluster": edge_to_col,
+        "geometry": "original_geometry",
+        "new_geometry": "geometry",
+    }
+    cols_drop = ["new_origin_pt", "new_destination_pt"]
+    new_edges_gdf = edges_simplified_gdf.rename(cols_rename, axis=1).drop(
+        columns=cols_drop
+    )
+    new_edges_gdf.loc[:, "length"] = new_edges_gdf.length
+
+    # Update the indices:
+    new_edges_gdf.loc[:, edge_from_col] = new_edges_gdf[edge_from_col].where(
+        new_edges_gdf[edge_from_col] >= 0, new_edges_gdf["original_from"]
+    )
+    new_edges_gdf.loc[:, edge_to_col] = new_edges_gdf[edge_to_col].where(
+        new_edges_gdf[edge_to_col] >= 0, new_edges_gdf["original_to"]
+    )
+
+    # Get the edge list with (from, to, data):
+    new_edges_list = list(
+        zip(
+            new_edges_gdf[edge_from_col],
+            new_edges_gdf[edge_to_col],
+            new_edges_gdf.iloc[:, 2:].to_dict("index").values(),
+        )
+    )
+
+    return new_edges_list
+
+
+def _extension_simplification(geometry, new_origin, new_destination):
+    """
+    Extends edge geometry to new endpoints.
+
+    If either new_origin or new_destination is None, maintains the
+      respective current endpoint.
+
+    Parameters
+    ----------
+    geometry : shapely.LineString
+    new_origin, new_destination: shapely.Point or None
```

Comment:
Should be two entries. But Since this is a non-user facing function maybe it doesn't need detailed docstrings? Maybe type hints though? @martinfleis What are your thoughts on this?

Reply:
for the sake of consistency I fixed in all cases